superset/reports/notifications/slack.py

```diff
@@ -60,16 +61,24 @@ class SlackNotification(BaseNotification):  # pylint: disable=too-few-public-met
 
     type = ReportRecipientType.SLACK
 
-    def _get_channel(self) -> str:
+    def _get_channels(self, client: WebClient) -> List[str]:
         """
         Get the recipient's channel(s).
-        Note Slack SDK uses "channel" to refer to one or more
-        channels. Multiple channels are demarcated by a comma.
-        :returns: The comma separated list of channel(s)
+        :returns: A list of channel ids: "EID676L"
         """
         recipient_str = json.loads(self._recipient.recipient_config_json)["target"]
 
-        return ",".join(get_email_address_list(recipient_str))
+        channel_recipients = get_email_address_list(recipient_str)
+
+        conversations_list_response = client.conversations_list(
+            types="public_channel,private_channel"
```

Comment:
We now need to map the list of names to a list of ids. This is the slack sdk method that fetches information that has both the name and id.

Reply:
Will this work if the application doesn't have the right scope? I tried testing this [here](https://api.slack.com/methods/conversations.list/test) with my token and got this response:

```
{
    "ok": false,
    "error": "missing_scope",
    "needed": "channels:read,groups:read,mpim:read,im:read",
    "provided": "commands,chat:write,chat:write.public,channels:history,im:history,mpim:history,groups:history,files:write"
}
```

Reply:
oh, that's a good point. @Vitor-Avila had a great solution which is to try the deprecated method first and then try the new one, since only new apps are affected right now. I'm also nervous about introducing a breaking change here.

Reply:
I assume right now we store the names of the channels, which is why we're mapping back to IDs? In the future it would be nice to store the IDs directly, not only it would eliminate the need for this mapping but also reports would keep working when channels are renamed.

Reply:
Yeah, correct, since that is what is in the report form. If we stored the ids, then we would need to map then back to the name when someone tries to view/edit the form, but that's prob still better.

Reply:
I was thinking about this more @betodealmeida and it would be a breaking change because the existing applications would need the `channels:read` slack scope for this, but maybe for the next major release, it would a nice improvement to have a multiselect ui for the channel and then also store the id. ❤️